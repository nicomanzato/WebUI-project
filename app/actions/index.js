export const TIMELINE_AVAILABLE = 'TIMELINE_AVAILABLE';
export const TIMELINE_APPEND_POST = 'TIMELINE_APPEND_POST';
export const TIMELINE_LOADING_MORE_POST = 'TIMELINE_LOADING_MORE_POST';
export const START_REFRESHING_TIMELINE = 'START_REFRESHING_TIMELINE';
export const FINISH_REFRESHING_TIMELINE = 'FINISH_REFRESHING_TIMELINE';

const serverIP = '192.168.0.111:8080';

export function getTimeline(configuration){
  return (dispatch) => {

    const count = 20;
    const url = `http://${serverIP}/timeline?count=${count}`;

    fetch(url)
      .then((response) => response.json())
      .then((responseJson) => {
        const data = applyTimelineConfigurations(responseJson, configuration);;
        dispatch({type: TIMELINE_AVAILABLE, data: data});
      })
      .catch((error) => {
        console.error(error);
      });

  }
}

export function fetchMoreTimeline(maxID, configuration){
  return (dispatch) => {

    const count = 20;
    const url = `http://${serverIP}/timeline?count=${count}&max_id=${maxID}`;

    fetch(url)
      .then((response) => response.json())
      .then((responseJson) => {
        const data = applyTimelineConfigurations(responseJson, configuration);
        dispatch({type: TIMELINE_APPEND_POST, data: data.slice(1)}); // drop first result which is already loaded.
      })
      .catch((error) => {
        console.error(error);
      });
  }
}

export function startLoadingMorePost(){
  return (dispatch) => {
    dispatch({type: TIMELINE_LOADING_MORE_POST});
  }
}

export function startRefreshingTimeline(){
  return (dispatch) => {
    dispatch({type: START_REFRESHING_TIMELINE});
  }
}

export function finishRefreshingTimeline(){
  return (dispatch) => {
    dispatch({type: FINISH_REFRESHING_TIMELINE});
  }
}

function applyTimelineConfigurations(newPosts, configuration) {

  let posts = newPosts;

  console.log("applying configurations");

  if (configuration.configVerifiedOnly) {
    console.log("removing tweets from not verified people");
    posts = posts.filter((post) => post.user.verified);
  }

  if (configuration.configDoNotFollow) {
    console.log("removing tweets from people you don't follow");
    posts = posts.filter((post) => post.user.following);
  }

  if (configuration.configHaveDefaultInformation) {
    console.log("removing tweets from people with default information");
    posts = posts.filter((post) => !post.user.default_profile);
  }

  if (configuration.configTextTruncated) {
    console.log("removing truncated tweets");
    posts = posts.filter((post) => post.entities.following);
  }

  if (configuration.configContainsLink) {
    console.log("removing tweets that contain links");
    console.log(posts.length);
    posts = posts.filter((post) => !new RegExp("([a-zA-Z0-9]+://)?([a-zA-Z0-9_]+:[a-zA-Z0-9_]+@)?([a-zA-Z0-9.-]+\\.[A-Za-z]{2,4})(:[0-9]+)?(/.*)?").test(post.text));
    console.log(posts.length);
  }

  return posts;
}
