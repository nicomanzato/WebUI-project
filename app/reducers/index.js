import { combineReducers } from 'redux';
import timelineReducer from './timelineReducer';
import nav from './navReducer';

const AppReducer = combineReducers({
  nav,
  timelineReducer
});

export default AppReducer;
