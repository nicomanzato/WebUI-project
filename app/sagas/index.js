import { call, all, put, takeEvery, takeLatest, take, select } from 'redux-saga/effects'
import { delay } from 'redux-saga'

import timelineSagas from './timelineSagas'
import trendSagas from './trendSagas'
import searchSagas from './searchSagas'

const serverIP = '10.160.11.56:8080';

export default function* rootSaga() {
  yield all([
    searchSagas(),
    timelineSagas(),
    trendSagas(),
  ])
}
