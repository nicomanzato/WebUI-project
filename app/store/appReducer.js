'use strict';

import { combineReducers } from 'redux'
import PostReducer from './post/postReducer'
import TrendReducer from './trend/trendReducer'
import UserReducer from './user/userReducer'
import nav from './navigation/navReducer'
import ConfigurationReducer from './configuration/configurationReducer'

const AppReducer = combineReducers({
  PostReducer,
  TrendReducer,
  UserReducer,
  nav,
  ConfigurationReducer,
});

export default AppReducer;
