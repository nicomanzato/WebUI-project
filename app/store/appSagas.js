import { call, all, put, takeEvery, takeLatest, take, select } from 'redux-saga/effects'
import { delay } from 'redux-saga'

import postSagas from './post/postSagas'
import trendSagas from './trend/trendSagas'
import userSagas from './user/userSagas'

export default function* appSagas() {
  yield all([
    postSagas(),
    trendSagas(),
    userSagas(),
  ])
}
