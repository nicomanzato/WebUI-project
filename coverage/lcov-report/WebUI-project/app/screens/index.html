<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for WebUI-project/app/screens</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> WebUI-project/app/screens
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">32.39% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>23/71</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">17.86% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/28</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">7.14% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/42</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">32.39% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>23/71</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="ConfigScreen.js"><a href="ConfigScreen.js.html">ConfigScreen.js</a></td>
	<td data-value="41.67" class="pic low"><div class="chart"><div class="cover-fill" style="width: 41%;"></div><div class="cover-empty" style="width:59%;"></div></div></td>
	<td data-value="41.67" class="pct low">41.67%</td>
	<td data-value="12" class="abs low">5/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	<td data-value="41.67" class="pct low">41.67%</td>
	<td data-value="12" class="abs low">5/12</td>
	</tr>

<tr>
	<td class="file medium" data-value="HomeScreen.js"><a href="HomeScreen.js.html">HomeScreen.js</a></td>
	<td data-value="55.56" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 55%;"></div><div class="cover-empty" style="width:45%;"></div></div></td>
	<td data-value="55.56" class="pct medium">55.56%</td>
	<td data-value="9" class="abs medium">5/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="55.56" class="pct medium">55.56%</td>
	<td data-value="9" class="abs medium">5/9</td>
	</tr>

<tr>
	<td class="file low" data-value="PostScreen.js"><a href="PostScreen.js.html">PostScreen.js</a></td>
	<td data-value="25" class="pic low"><div class="chart"><div class="cover-fill" style="width: 25%;"></div><div class="cover-empty" style="width:75%;"></div></div></td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="8" class="abs low">2/8</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="8" class="abs low">2/8</td>
	</tr>

<tr>
	<td class="file low" data-value="SearchScreen.js"><a href="SearchScreen.js.html">SearchScreen.js</a></td>
	<td data-value="9.38" class="pic low"><div class="chart"><div class="cover-fill" style="width: 9%;"></div><div class="cover-empty" style="width:91%;"></div></div></td>
	<td data-value="9.38" class="pct low">9.38%</td>
	<td data-value="32" class="abs low">3/32</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="18" class="abs low">0/18</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="19" class="abs low">0/19</td>
	<td data-value="9.38" class="pct low">9.38%</td>
	<td data-value="32" class="abs low">3/32</td>
	</tr>

<tr>
	<td class="file high" data-value="UserProfileScreen.js"><a href="UserProfileScreen.js.html">UserProfileScreen.js</a></td>
	<td data-value="80" class="pic high"><div class="chart"><div class="cover-fill" style="width: 80%;"></div><div class="cover-empty" style="width:20%;"></div></div></td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="10" class="abs high">8/10</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="42.86" class="pct low">42.86%</td>
	<td data-value="7" class="abs low">3/7</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="10" class="abs high">8/10</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Dec 28 2018 10:06:03 GMT-0300 (GMT-03:00)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
