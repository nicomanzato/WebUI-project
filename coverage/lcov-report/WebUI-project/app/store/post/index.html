<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for WebUI-project/app/store/post</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\index.html">All files</a> WebUI-project/app/store/post
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">92.02% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>173/188</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">97.56% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>40/41</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">97.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>42/43</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">91.8% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>168/183</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="postActions.js"><a href="postActions.js.html">postActions.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="57" class="abs high">57/57</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="57" class="abs high">57/57</td>
	</tr>

<tr>
	<td class="file high" data-value="postReducer.js"><a href="postReducer.js.html">postReducer.js</a></td>
	<td data-value="95.24" class="pic high"><div class="chart"><div class="cover-fill" style="width: 95%;"></div><div class="cover-empty" style="width:5%;"></div></div></td>
	<td data-value="95.24" class="pct high">95.24%</td>
	<td data-value="42" class="abs high">40/42</td>
	<td data-value="95.24" class="pct high">95.24%</td>
	<td data-value="21" class="abs high">20/21</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="95.24" class="pct high">95.24%</td>
	<td data-value="42" class="abs high">40/42</td>
	</tr>

<tr>
	<td class="file medium" data-value="postSagas.js"><a href="postSagas.js.html">postSagas.js</a></td>
	<td data-value="79.03" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 79%;"></div><div class="cover-empty" style="width:21%;"></div></div></td>
	<td data-value="79.03" class="pct medium">79.03%</td>
	<td data-value="62" class="abs medium">49/62</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="93.75" class="pct high">93.75%</td>
	<td data-value="16" class="abs high">15/16</td>
	<td data-value="79.03" class="pct medium">79.03%</td>
	<td data-value="62" class="abs medium">49/62</td>
	</tr>

<tr>
	<td class="file high" data-value="postSelector.js"><a href="postSelector.js.html">postSelector.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="27" class="abs high">27/27</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="20" class="abs high">20/20</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Dec 27 2018 17:02:42 GMT-0300 (GMT-03:00)
</div>
</div>
<script src="..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\sorter.js"></script>
</body>
</html>
