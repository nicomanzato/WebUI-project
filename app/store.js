import { applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga'
import appReducer from './store/appReducer'; //Import the reducer
import appSagas from './store/appSagas';
import reactotron from '../reactotronConfig'

const sagaMonitor = reactotron.createSagaMonitor()


const sagaMiddleware = createSagaMiddleware({ sagaMonitor })

export const store = reactotron.createStore(appReducer, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(appSagas)
