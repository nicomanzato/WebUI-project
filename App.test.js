import React from 'react';
import {App} from './app'
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import {store} from './app/store'

describe('app', () => {
  it('should render correctly', () => {
    const mockNavigationProp = {state: {params: {data: '1'}}}
    const tree = renderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
