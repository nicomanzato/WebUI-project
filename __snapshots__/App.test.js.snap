// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should render correctly 1`] = `
<View
  collapsable={false}
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    collapsable={undefined}
    style={
      Object {
        "backgroundColor": "white",
        "elevation": 4,
        "justifyContent": "center",
        "padding": 0,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 0.5,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 0.5,
        "zIndex": 1,
      }
    }
  >
    <View
      collapsable={undefined}
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "backgroundColor": "#ffeb3b",
            "borderBottomColor": "#1183ff",
            "borderBottomWidth": 2,
            "bottom": 0,
            "height": 2,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "width": 0,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "overflow": "scroll",
        }
      }
    >
      <RCTScrollView
        alwaysBounceHorizontal={false}
        automaticallyAdjustContentInsets={false}
        bounces={undefined}
        collapsable={undefined}
        contentContainerStyle={
          Array [
            Object {
              "flexDirection": "row",
              "flexWrap": "nowrap",
            },
            Object {
              "flex": 1,
            },
          ]
        }
        contentOffset={undefined}
        horizontal={true}
        keyboardShouldPersistTaps="handled"
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        overScrollMode="never"
        scrollEnabled={undefined}
        scrollEventThrottle={1}
        scrollsToTop={false}
        showsHorizontalScrollIndicator={false}
      >
        <View>
          <View
            accessibilityComponentType="button"
            accessibilityLabel="Home"
            accessibilityTraits={
              Array [
                "button",
                "selected",
              ]
            }
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={undefined}
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "justifyContent": "center",
                    "opacity": 1,
                    "padding": 12,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": 24,
                        "width": 24,
                      },
                      Object {
                        "height": 35,
                        "marginTop": 0,
                        "padding": 0,
                        "width": 35,
                      },
                    ]
                  }
                >
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "height": "100%",
                        "justifyContent": "center",
                        "minWidth": 25,
                        "opacity": 1,
                        "position": "absolute",
                        "width": "100%",
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    />
                  </View>
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "height": "100%",
                        "justifyContent": "center",
                        "minWidth": 25,
                        "opacity": 0,
                        "position": "absolute",
                        "width": "100%",
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            accessibilityComponentType="button"
            accessibilityLabel="Search"
            accessibilityTraits="button"
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={undefined}
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "justifyContent": "center",
                    "opacity": 0.7,
                    "padding": 12,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": 24,
                        "width": 24,
                      },
                      Object {
                        "height": 35,
                        "marginTop": 0,
                        "padding": 0,
                        "width": 35,
                      },
                    ]
                  }
                >
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "height": "100%",
                        "justifyContent": "center",
                        "minWidth": 25,
                        "opacity": 0,
                        "position": "absolute",
                        "width": "100%",
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    />
                  </View>
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "height": "100%",
                        "justifyContent": "center",
                        "minWidth": 25,
                        "opacity": 1,
                        "position": "absolute",
                        "width": "100%",
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            accessibilityComponentType="button"
            accessibilityLabel="Config"
            accessibilityTraits="button"
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={undefined}
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "justifyContent": "center",
                    "opacity": 0.7,
                    "padding": 12,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": 24,
                        "width": 24,
                      },
                      Object {
                        "height": 35,
                        "marginTop": 0,
                        "padding": 0,
                        "width": 35,
                      },
                    ]
                  }
                >
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "height": "100%",
                        "justifyContent": "center",
                        "minWidth": 25,
                        "opacity": 0,
                        "position": "absolute",
                        "width": "100%",
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    />
                  </View>
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "height": "100%",
                        "justifyContent": "center",
                        "minWidth": 25,
                        "opacity": 1,
                        "position": "absolute",
                        "width": "100%",
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      alwaysBounceHorizontal={false}
      automaticallyAdjustContentInsets={false}
      bounces={false}
      contentContainerStyle={
        Object {
          "flex": 1,
        }
      }
      contentOffset={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      directionalLockEnabled={true}
      horizontal={true}
      keyboardDismissMode="on-drag"
      keyboardShouldPersistTaps="always"
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={undefined}
      onScrollEndDrag={undefined}
      overScrollMode="never"
      pagingEnabled={true}
      scrollEnabled={undefined}
      scrollEventThrottle={1}
      scrollsToTop={false}
      showsHorizontalScrollIndicator={false}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            Object {
              "flex": 1,
              "overflow": "hidden",
            }
          }
          testID={undefined}
        >
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            <View
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column-reverse",
                  },
                  Object {
                    "backgroundColor": "#000",
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  collapsable={undefined}
                  pointerEvents="auto"
                  style={
                    Object {
                      "backgroundColor": "#E9E9EF",
                      "bottom": 0,
                      "left": 0,
                      "marginTop": -0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "shadowColor": "black",
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0.2,
                      "shadowRadius": 5,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#FFFF",
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#fff",
                          "flex": 1,
                          "justifyContent": "center",
                        }
                      }
                    >
                      <ActivityIndicator
                        animating={true}
                        color="#999999"
                        hidesWhenStopped={true}
                        size="small"
                      />
                    </View>
                  </View>
                </View>
              </View>
              <View
                onLayout={[Function]}
                pointerEvents="box-none"
              >
                <View
                  collapsable={undefined}
                  style={
                    Object {
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    collapsable={undefined}
                    onLayout={[Function]}
                    pointerEvents="box-none"
                    style={
                      Object {
                        "backgroundColor": "#F7F7F7",
                        "borderBottomColor": "#A7A7AA",
                        "borderBottomWidth": 0.5,
                        "height": 64,
                        "paddingBottom": 0,
                        "paddingLeft": 0,
                        "paddingRight": 0,
                        "paddingTop": 20,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    />
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          style={null}
          testID={undefined}
        />
        <View
          accessibilityElementsHidden={true}
          importantForAccessibility="no-hide-descendants"
          style={null}
          testID={undefined}
        />
      </View>
    </RCTScrollView>
  </View>
</View>
`;
